src/dynamicResources/awsResourceManager.ts

```diff
@@ -85,13 +85,15 @@ export class AwsResourceManager {
         }
     }
 
-    public async close(uri: vscode.Uri): Promise<void> {
+    public async close(uri: vscode.Uri, isEditorClosed?: boolean): Promise<void> {
         const path = uri.toString()
         if (this.openResources.has(path)) {
             getLogger().debug(`resourceManager: closing ${uri}`)
 
-            await vscode.window.showTextDocument(uri)
-            await vscode.commands.executeCommand('workbench.action.closeActiveEditor')
+            if (!isEditorClosed) {
+                await vscode.window.showTextDocument(uri)
```

Comment:
Why are we "showing" the document in close()? Worth a 1-line comment above this line to help future readers.